### PR TITLE
Refactor remark to project

### PR DESCRIPTION
**Pull request by antonTan96**
Also Implemented:
Storing projects in Taskhub
addP (Usage: addP pr/PROEJCT_NAME e/1 2 3 5 4...)

Updating TaskHub to support projects caused some tests that used the old TaskHub to fail.
Test cases regarding new code are also omitted, the test cases will be written concurrently while trying to incorporate projects into the GUI of TaskHub.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public class AddCommandParser implements Parser<AddCommand> {
     public AddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
-                        PREFIX_REMARK, PREFIX_TAG);
+                         PREFIX_TAG, PREFIX_PROJECT);
```
Comment (aslam341) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:37]: For standardization, I think `PREFIX_PROJECT` should come before `PREFIX_TAG` as follows:
`ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
                         PREFIX_PROJECT, PREFIX_TAG);`

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -10,7 +10,8 @@ public class CliSyntax {
     public static final Prefix PREFIX_PHONE = new Prefix("p/");
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
-    public static final Prefix PREFIX_REMARK = new Prefix("r/");
+    public static final Prefix PREFIX_PROJECT = new Prefix("pr/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
+    public static final Prefix PREFIX_EMPLOYEE = new Prefix("e/");
```
Comment (aslam341) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:15]: Does this not clash with the PREFIX_EMAIL, as they both use the the same Prefix "e/".

**Review by aslam341 — CHANGES_REQUESTED** [review]
Just a minor change and question. Also, you'll need to add more tests as highlighted by codecov.

File: src/main/java/seedu/address/model/employee/Project.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Project // instanceof handles nulls
+                && name.equals(((Project) other).name)); // state check
+    }
+
```
Comment (Chandan8186) [inline src/main/java/seedu/address/model/employee/Project.java:98]: From what I understand here, isSameProject method and equals method are doing the same thing. Perhaps isSameProject can be removed if equals method can be used in place of it.

**Review by Chandan8186 — CHANGES_REQUESTED** [review]
Just a minor comment about the Project class.

File: src/main/java/seedu/address/model/employee/Project.java (hunk 3 above)
Comment (Chandan8186) [reply src/main/java/seedu/address/model/employee/Project.java:98]: I'd like to add to the above comment, I feel it is best to keep `isSameProject` method, but amend it such that it compares the employees listed under the project (and any other project details), on top of just the project name. This is in line with `isSameEmployee` and `equals` method of the `Employee` class.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 2 above)
Comment (antonTan96) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:15]: Yes, but unlike the previous case where project and phone could exist in the add command, there hypothetically shouldn't be a command that needs both prefixes (edit command uses initial index as employee)

**Pull request closed without merging**